packages/components/src/color-picker/test/index.tsx

```diff
@@ -13,20 +13,19 @@ import { ColorPicker } from '..';
  * on several elements and we'd end up encoding assumptions about order when
  * trying to select the appropriate element. We might as well just use the class name
  * on the container which will be more durable if, for example, the order changes.
- *
- * @param {HTMLElement} container
- * @return {HTMLElement} The saturation element
  */
-function getSaturation( container ) {
+function getSaturation( container: HTMLElement ) {
 	return container.querySelector(
 		'.react-colorful__saturation .react-colorful__interactive'
 	);
 }
 
+type PageXPageY = { pageX: number; pageY: number };
+
 // Fix to pass `pageX` and `pageY`
 // See https://github.com/testing-library/react-testing-library/issues/268
-class FakeMouseEvent extends window.MouseEvent {
-	constructor( type, values = {} ) {
+class FakeMouseEvent extends MouseEvent {
+	constructor( type: MouseEvent[ 'type' ], values: PageXPageY ) {
```

Comment:
Good point, thank you.

I needed to make some small changes alongside this to keep TS happy: 87ca35d8eb

Reply:
I took another look at this, and I actually think that your initial approach of removing the default `{}` value may be actually be cleaner — I would just tweak it so that `values` can be optional (which is probably the reason why we had that default value of `{}`).

We can greatly simplify the code by using optional chaining:

```diff
diff --git a/packages/components/src/color-picker/test/index.tsx b/packages/components/src/color-picker/test/index.tsx
index addda8c6b4..9ef79cde2e 100644
--- a/packages/components/src/color-picker/test/index.tsx
+++ b/packages/components/src/color-picker/test/index.tsx
@@ -25,16 +25,12 @@ type PageXPageY = { pageX: number; pageY: number };
 // Fix to pass `pageX` and `pageY`
 // See https://github.com/testing-library/react-testing-library/issues/268
 class FakeMouseEvent extends MouseEvent {
-	constructor( type: MouseEvent[ 'type' ], values: PageXPageY | {} = {} ) {
+	constructor( type: MouseEvent[ 'type' ], values?: PageXPageY ) {
 		super( type, { buttons: 1, bubbles: true, ...values } );
 
-		const parseXYValue = ( value: number ) => {
-			return value ?? 0;
-		};
-
 		Object.assign( this, {
-			pageX: 'pageX' in values ? parseXYValue( values.pageX ) : 0,
-			pageY: 'pageY' in values ? parseXYValue( values.pageY ) : 0,
+			pageX: values?.pageX ?? 0,
+			pageY: values?.pageY ?? 0,
 		} );
 	}
 }

```

Reply:
Yes, that's much better. b0e9423679